### DISCUSSION TITLE
How much does it cost

### DISCUSSION THREAD

**Discussion by Xyndra** [Pythagora-io/gpt-pilot#76] [Q&A]
GPT-4 is hella expensive if it gets out of control, so how much do you usually need to pay to use it?

**Comment by solomonboltin-mykey** [comment]
About 2-3 dollars for a small project

**Comment by YosuCadilla** [comment]
TL;DR: My first project costed $33. Next time, I would expect to cut cost down by 50% and work hours by 75%

This was the cost for a medium-low sized project.
About 50 hours of active work and 2,991,885 tokens (about $33)

CONGRATULATIONS!!!
You reached 100% of your project generation!
For now, you have created:
17 files
629 lines of code
Before continuing, GPT Pilot will create some documentation for the project...
Creating README.md

This was the first time I attempted any serious development with GPTP, so much of the time went to actually learning how GPTP works and setting up the environments, as well as writing and optimizing prompts (project descriptions).
Making it run satisfactorily required over 15 iterations on the same project with different prompts.
That is why, the next time I run a GPTP project, I would expect to cut cost down by 50% and work hours by 75%.

**Reply by nicoleCodeGirl** [reply]
I downloaded Pythagora and it started creating an app for me and right before it completed it it said my trial expired. What a teaser! And I'm not trying to pay for something that I'm not sure if it really works? Also is there even a free version cause the YouTube video I watched said it was free

**Reply by Kelvinm** [reply]
Don't forget now even to sign up you're forced to consent to their marketing spam.

They have online that its free, but it seems misleading.

**Reply by tobiaswaggoner** [reply]
This "free trial ended" confused me also.

All you have to do is provide your own OpenAI API key in the .env (https://github.com/Pythagora-io/gpt-pilot/wiki/Using-Pythagora-with-your-own-OpenAI-key) and it will happily continue to generate.

If you do not provide your own OpenAI API key, then you are using their keys and hence they pay for the inference. These costs are then basically passed on to you, alas this is not documented well.

I think the business model is valid. 

The main tool is open source, but by providing a hosted endpoint they can improve the pipeline and do other stuff not least of all see if and how the tool is really used.

Also the users do not have to set up accounts for every (future) supported model - they can route the requests.

**Comment by mycryptosuccess** [comment]
im confused as to where to find the .env file to edit and add my key
